_posts/2019-02-08-resize-images-on-the-fly.md

```diff
@@ -0,0 +1,187 @@
+
+Shortly after that we launched our v2 version of our platform where we came up with a use case in which we need to give users a feature to download and resize images on click. We also wanted to show thumbnails in different sizes dynamically. In this article, I’ll show you how we’ve implemented resizing the images on the fly with Node.js and deploy it with OpenFaaS.
+Where do we start?
+
+## Pre-requisites for the tutorial
+
+You’ll need:
+
+* [Docker or Docker Desktop](https://www.docker.com/products/docker-desktop)
+* OpenFaaS
+* faas-cli
+* [`npm` and `node`](https://nodejs.org/en/download/)
+
+If you haven’t got OpenFaaS yet, then start here - it only takes a few minutes https://docs.openfaas.com/deployment/ 
+
+## Create a new Node.js function
+
+So, let’s create our function to resize images, we’ll call it image-resize. Instead of the classic Node template, I’ll be using the node10-express template because it offers full control over HTTP headers and higher throughput. That means that we can set the Content-Type when resizing different types of images for our users.
+
+1. Create a new Function
+
+Create a new folder for our work
+
+```bash
+$ mkdir -p ~/dev/resizer \
+&& cd ~/dev/resizer
+```
+
+Now, pull the node10-express template from the OpenFaaS Template Store.
+
+```bash
+$ faas-cli template store pull node10-express
+```
+
+Create new function from the template
+
+```bash
+$ faas-cli new image-resize --lang=node10-express --prefix=USERNAME
+```
+
+> Note: Replace USERNAME, with your name on the Docker Hub, or your private registry address.
+
+This will create a new folder named image-resize having handler.js and package.json in it.
+
+```bash
+image-resize/handler.js
+image-resize/package.json
+image-resize.yml
+```
+
+The newly generated function is a hello-world style example which we can edit in a moment.
+
+Let’s deploy it and try it out.
+
+```bash
+$ faas-cli up -f image-resize.yml
+```
+
+We’ll check the deployed function. Then we will continue to write our image resizing functionality.
+
+(Image here)
+
+The default sample function serving traffic in the OpenFaaS Portal.
+
+2. Write the resizing logic using sharp
+
+We are going to use the sharp package to resize the images. This is where we would have normally used ImageMagick or GraphicsMagick. This is what the sharp website says about the library:
+
+> The typical use case for this high speed Node.js module is to convert large images in common formats to smaller, web-friendly JPEG, PNG and WebP images of varying dimensions.
+> Resizing an image is typically 4x-5x faster than using the quickest ImageMagick and GraphicsMagick settings.
+
+So at IconScout, having a 4x-5x speed increase was a great move forward for us where we process 10s of thousands of images per day and that’s growing every week.
+
+Now let’s install the sharp package using `npm`.
+
+```bash
+$ cd ~/dev/resizer/image-resizer
```

Comment:
The function name changes here from `image-resize` to `image-resizer`
```suggestion
$ cd ~/dev/resizer/image-resize
```

Reply:
👍